packages/appinio_swiper/lib/appinio_swiper.dart

```diff
@@ -1,679 +1,743 @@
+import 'dart:collection';
 import 'dart:math';
 
 import 'package:appinio_swiper/types.dart';
 import 'package:flutter/material.dart';
 
-import 'controllers.dart';
 import 'enums.dart';
 
-export 'controllers.dart';
 export 'enums.dart';
 export 'types.dart';
 
 class AppinioSwiper extends StatefulWidget {
-  /// when the user cancels the swipe
-
+  /// A callback that's called when the user lifts their finger during a drag
+  /// without having reached [threshold].
   final VoidCallback? onSwipeCancelled;
 
-  /// widget builder for creating cards
-  final CardsBuilder cardsBuilder;
+  /// The indexed widget builder that builds a card for the given index.
+  final IndexedWidgetBuilder cardsBuilder;
 
-  ///cards count
-  final int cardsCount;
+  /// The number of cards in the stack.
+  final int cardCount;
 
-  ///cards spacing
+  /// Cards spacing.
   final double cardsSpacing;
 
-  ///background cards count
-  final int backgroundCardsCount;
-
-  /// controller to trigger unswipe action
+  /// Background cards count
+  final int backgroundCardCount;
+
+  /// The amount to scale each successive background card down by,
+  ///
+  /// The difference in scale for each background card is a fixed amount
+  /// relative to the original card.
+  ///
+  /// Defaults to .9.
+  final double backgroundCardScale;
+
+  /// The amount to offset each successive background card relative to the card
+  /// before it.
+  ///
+  /// Defaults to offsetting each card down by 40 dp.
+  final Offset? backgroundCardOffset;
+
+  /// A controller that provides programmatic control of the swiper and notifies
+  /// on swiper state changes.
   final AppinioSwiperController? controller;
 
-  /// duration of every animation
+  /// The duration of swipe animations.
+  ///
+  /// Swipe animations start after the user lifts their finger or when a drag
+  /// is triggered by [AppinioSwiperController].
   final Duration duration;
 
-  /// add listener to check when the card is swiping
-  final void Function(AppinioSwiperDirection direction)? onSwiping;
+  /// A callback that is called whenever the card moves.
+  final void Function(AxisDirection direction, Offset position)?
+      onCardPositionChanged;
 
-  /// padding of the swiper
-  final EdgeInsetsGeometry padding;
```

Comment:
I wasn't sure why this was here. The dev can just wrap the swiper in their own padding widget, unless I'm missing something here.